File: leetcode.js/对称二叉树.js
var isSymmetric = function(root) {
    if(!root) return true;
    const ismirror = (left,right) => {
        if(!left && !right) return true;
        if(!left || !right) return false;
        if(left.val !== right.val) return false;
        return ismirror(left.left , right.right) && ismirror(left.right , right.left);
    }
    return ismirror(root.left,root.right);

};
// 示例用法
// 定义二叉树节点类
class TreeNode {
    constructor(val = 0, left = null, right = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

// 构建对称二叉树
const symmetricTree = new TreeNode(1);
symmetricTree.left = new TreeNode(2);
symmetricTree.right = new TreeNode(2);
symmetricTree.left.left = new TreeNode(3);
symmetricTree.left.right = new TreeNode(4);
symmetricTree.right.left = new TreeNode(4);
symmetricTree.right.right = new TreeNode(3);

// 构建非对称二叉树
const nonSymmetricTree = new TreeNode(1);
nonSymmetricTree.left = new TreeNode(2);
nonSymmetricTree.right = new TreeNode(2);
nonSymmetricTree.left.right = new TreeNode(3);
nonSymmetricTree.right.right = new TreeNode(3);

console.log(isSymmetric(symmetricTree)); // 输出: true
console.log(isSymmetric(nonSymmetricTree)); // 输出: false

File: leetcode.py/对称二叉树.py
class Solution(object):
    def isSymmetric(self, root):
        if not root:
            return True
        def isMirror(left,right):
            if(not left and not right):
                return True
            if(not left or not right):
                return False
            if(left.val != right.val):
                return False
            return isMirror(left.left,right.right)  and isMirror(left.right,right.left)
        # 以下为创建实例并调用的示例
        # 首先定义二叉树节点类
        class TreeNode:
            def __init__(self, val=0, left=None, right=None):
                self.val = val
                self.left = left
                self.right = right
        # 构建一个对称二叉树实例
        #       1
        #      / \
        #     2   2
        #    / \ / \
        #   3  4 4  3
        root = TreeNode(1)
        root.left = TreeNode(2)
        root.left.left = TreeNode(3)
        root.left.right = TreeNode(4)
        root.right = TreeNode(2)
        root.right.left = TreeNode(4)
        root.right.right = TreeNode(3)
        # 调用 isMirror 函数判断是否对称
        result = isMirror(root.left, root.right)
        return result
